### DISCUSSION TITLE
PSA: Kryoflux SW will blacklist cdc_acm and break GW on Linux

### DISCUSSION THREAD

**Discussion by wohali** [keirf/greaseweazle#312] [General]
Seen elsewhere. Thought others should know.

![signal-2023-05-06-212738](https://user-images.githubusercontent.com/112292/236656575-caba3173-6ee0-4f0f-84d7-4b66c749d129.jpeg)

**Comment by SimonGoodwin** [comment]
Greaseweasel is very poorly documented and treats anyone not running Windows as an afterthought, at best. This makes it hard to recommend the hardware to anyone else, which is a shame.

**Reply by ewanuno** [reply]
Luckily it's open source so submissions of improved documentation is always possible.

Open source software isn't just normal software that you don't have to pay for, it's a collaborative process that you can become involved in if you wish to change it.

If you want software that you can make demands to the authors, and feel aggrieved if it isn't the way you want it, then there are other projects that offer that service. They all cost money.

**Reply by keirf** [reply]
ITYM "Kryoflux". Or I certainly HYM.

**Comment by emersonrp** [comment]
In here over a year later to thank you for this info -- I ran into this just now.

A little digging turns up that the cdc_acm module actually kernel panics if it finds the Kryoflux hardware, hence the AUR package blacklisting the module.  This has been a known thing since at least 2010, but I haven't seen any motion on this either with Kryoflux or the Linux USB people since 2012, so it might just Be That Way.  Someone might lead the charge to work out the problem so that the blacklist's not necessary any more, but I don't feel qualified to be that person.
